django/urls/resolvers.py

```diff
@@ -315,24 +315,38 @@ class RoutePattern(CheckURLMixin):
     regex = LocaleRegexRouteDescriptor()
 
     def __init__(self, route, name=None, is_endpoint=False):
-        self._route = route
-        self._regex, self.converters = _route_to_regex(str(route), is_endpoint)
+        self._route = str(route)
```

Comment:
I think this is not safe for translations. The descriptor at `self.regex` goes through translations. Also, we would still want to apply translations in the new fast path. The cast to string can be done there.